src/index.ts

```diff
@@ -327,6 +330,53 @@ app.get('/api/advertising-info', (_req, res) => {
 
 app.use('/api/slise-ad-rules', sliseRulesRouter);
 
+app.post('/api/magic-square-quest/start', async (req, res) => {
+  try {
+    await startMagicSquareQuest(req.body);
+
+    res.status(200).send({ message: 'Quest successfully started' });
+  } catch (error: any) {
+    console.error(error);
+
+    if (error instanceof CodedError) {
+      res.status(error.code).send(error.buildResponse());
+    } else {
+      res.status(500).send({ message: error?.message });
+    }
+  }
+});
+
+app.get('/api/magic-square-quest/participants', basicAuth, async (req, res) => {
+  try {
+    res.status(200).send(await getMagicSquareQuestParticipants());
+  } catch (error: any) {
+    console.error(error);
+
+    if (error instanceof CodedError) {
+      res.status(error.code).send(error.buildResponse());
+    } else {
+      res.status(500).send({ message: error?.message });
+    }
+  }
+});
+
+app.get('/api/signing-nonce', (req, res) => {
+  try {
+    const pkh = req.query.pkh;
+    if (!pkh || typeof pkh !== 'string') throw new Error('PKH is not a string');
+
+    res.status(200).send(getSigningNonce(pkh));
+  } catch (error: any) {
+    console.error(error);
+
+    if (error instanceof CodedError) {
```

Comment:
We should consider moving this repetitive CodedError handling logic to a middleware